Pyjion/absint.cpp

```diff
@@ -2312,7 +2312,49 @@ JittedCode* AbstractInterpreter::compile_worker() {
                     auto two = stackInfo[stackInfo.size() - 2];
 
                     // Currently we only optimize floating point numbers..
-                    if (one.Value->kind() == AVK_Float && two.Value->kind() == AVK_Float) {
+                    if (one.Value->kind() == AVK_Integer && two.Value->kind() == AVK_Float) {
+                        _ASSERTE(m_stack[m_stack.size() - 1] == STACK_KIND_OBJECT);
+                        _ASSERTE(m_stack[m_stack.size() - 2] == STACK_KIND_VALUE);
+
+                        if (byte == BINARY_AND || byte == INPLACE_AND || byte == INPLACE_OR || byte == BINARY_OR ||
+                            byte == INPLACE_LSHIFT || byte == BINARY_LSHIFT || byte == INPLACE_RSHIFT || byte == BINARY_RSHIFT ||
+                            byte == INPLACE_XOR || byte == BINARY_XOR) {
+                            char buf[181];
+                            sprintf_s(buf, "unsupported operand type(s) for %s: 'float' and 'int'", op_to_string(byte));
+                            m_comp->emit_pyerr_setstring(PyExc_TypeError, buf);
+                            branch_raise();
+                            break;
+                        }
+                        if (byte == INPLACE_TRUE_DIVIDE || byte == BINARY_TRUE_DIVIDE ||
+                            byte == INPLACE_FLOOR_DIVIDE || byte == BINARY_FLOOR_DIVIDE ||
+                            byte == INPLACE_MODULO || byte == BINARY_MODULO) {
+                            m_comp->emit_dup();
+                            m_comp->emit_unary_not_tagged_int_push_bool();
```

Comment:
No, we know we have a tagged int or a real int...  I'll add some comments around this.

---
In reply to: [63258196](https://github.com/Microsoft/Pyjion/pull/186#discussion_r63258196) [](ancestors = 63258196)